### PR TITLE
metric1 unit tests

### PR DESCRIPTION
**Pull request by jelly**
After this PR, the cockpit-bridge and Python bridge should be somewhat consistent in error handling

### CODE REVIEW

File: pkg/base1/test-metrics.js [hunk 1]
```diff
@@ -104,6 +104,16 @@ function MockSink(expected, callback) {
     return self;
 }
 
+let old_channel;
+QUnit.module("Mock tests", {
+    beforeEach: () => {
+        old_channel = cockpit.channel;
+    },
+    afterEach: () => {
+        cockpit.channel = old_channel;
```
Comment (martinpitt) [inline pkg/base1/test-metrics.js:113]: Why is that necessary? AFAIK it does not get diverted anywhere?

File: pkg/base1/test-metrics.js [hunk 2]
```diff
@@ -127,4 +137,60 @@ QUnit.test("non-instanced decompression", function (assert) {
     assert.deepEqual(sink.samples, [[10], [10]], "got correct samples");
 });
 
+QUnit.test("interval validation", function (assert) {
```
Comment (martinpitt) [inline pkg/base1/test-metrics.js:140]: I'd rather we'd used ES6 `assert => {... }` syntax, but not a biggie.

File: pkg/base1/test-metrics.js [hunk 3]
```diff
@@ -127,4 +137,60 @@ QUnit.test("non-instanced decompression", function (assert) {
...
+    const metrics_channel = cockpit.channel({
+        payload: "metrics1",
+        source: "internal",
+        interval: -1,
+    });
+
+    metrics_channel.addEventListener("close", (_, error) => {
+        assert.equal(error.problem, "protocol-error");
+        assert.equal(error.message, 'invalid "interval" value: -1');
+        assert.ok(true, "Connection was closed for invalid interval");
```
Comment (martinpitt) [inline pkg/base1/test-metrics.js:152]: This assertion feels superfluous? We only need these if there's nothing more specific to assert, no?

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! Some cleanups (perhaps), but no strong "no". In case you push again, please add "no-test", as these unit tests don't run in the integration tests anyway.

File: pkg/base1/test-metrics.js (hunk 1 above)
Comment (jelly) [reply pkg/base1/test-metrics.js:113]: It does in mockChannel.

File: pkg/base1/test-metrics.js (hunk 3 above)
Comment (jelly) [reply pkg/base1/test-metrics.js:152]: Yeah, can be dropped.

File: pkg/base1/test-metrics.js (hunk 2 above)
Comment (jelly) [reply pkg/base1/test-metrics.js:140]: Can do, as I'm pushing anyway.

File: pkg/base1/test-metrics.js (hunk 1 above)
Comment (martinpitt) [reply pkg/base1/test-metrics.js:113]: Oh, you mean an existing test damages this, and for the new ones we want to use the real cockpit.channel.

**Review by martinpitt — APPROVED** [review]
Danke!

**Comment by jelly** [comment]
```
# Result testBasic (__main__.TestTeam.testBasic) failed
# 1 TEST FAILED [81s on ip-172-31-21-69]
```

Seems this one fails in all PR's

**Comment by martinpitt** [comment]
The failing test is `TestStorageUsed.testUsedAsPV`, that's obviously independent from this PR, but we need to investigate. It's very likely to come up at our next fedora-37 image refresh, too.
